main.go

```diff
@@ -198,6 +201,19 @@ func runWithOptions(opts Options, out output) {
 	})
 }
 
+type noEnveloper interface {
+	WithoutEnvelopes() encoding.Serializer
+}
+
+// withTransportSerializer may modify the serializer for the transport used.
+// E.g. Thrift payloads are not enveloped when used with TChannel.
+func withTransportSerializer(p transport.Protocol, s encoding.Serializer) encoding.Serializer {
+	if p == transport.TChannel && s.Encoding() == encoding.Thrift {
+		s = s.(noEnveloper).WithoutEnvelopes()
```

Comment:
Ok, so the owness is on TChannel to turn off envelopes, that makes a lot of sense 💯